### PR TITLE
Added messages rating(unstable), and running on GPU....

### PR DESCRIPTION
**Pull request by Jiankun-Huang**
Colab for running on GPU:
https://cdn.discordapp.com/attachments/1070838609051144212/1092617824494698628/gpt4all_gpu.ipynb

Rest of the code is a RUFT DRAFT!!

### CODE REVIEW

**Comment by ParisNeo** [comment]
Thanks. I'll look it up this evening (european time).

**Review by ParisNeo — CHANGES_REQUESTED** [review]
We need to make GPU usage optional.
You can add a command line option to select CPU or GPU (using argparse)

You are specifying the model path directly in the code. This is not possible. The user needs to be able to download and use the model. In the next version I am going to be using the new python bindings to this, so you may just keep it as it was until I finish this migration this evening (western european time).

I also prefer keeping the character by character function as it is way more user friendly. I'll take care of adding the message to the discussion when it ends.

If you want, just keep the old streaming way:
```
                bot = self.chatbot_bindings.bot
                bot.stdin.write(message.encode('utf-8'))
                bot.stdin.write(b"\n")
                bot.stdin.flush()

                # Segmented (the user receives the output as it comes)
                return Response(stream_with_context(self.parse_to_prompt_stream()))
```                
And keep the CPU version as I'm going to do a big update this evening on which you can build. GPU will be an option in command line.

Thank you for your help.

**Comment by Jiankun-Huang** [comment]
Specifying is gpu is simple, can do later

**Comment by Jiankun-Huang** [comment]
I can keep the old streaming too

**Comment by ParisNeo** [comment]
Can you please update your code so that I can merge it.
Consider keeping the GPU part as We are moving to a new python binding library maybe tomorrow
Best regards

File: static/js/chat.js [hunk 1]
```diff
@@ -225,22 +243,7 @@ function add_collapsible_div(text, id){
...
-<br>Examples:<br>
-<code>
-- A color description has been provided. Find the CSS code associated with that color. A light red color with a medium light shade of pink.<br>
-- Come up with an interesting idea for a new movie plot. Your plot should be described with a title and a summary.<br>
-- Reverse a string in python.<br>
-- List 10 dogs.<br>
-- Write me a poem about the fall of Julius Ceasar into a ceasar salad in iambic pentameter.<br>
-- What is a three word topic describing the following keywords: baseball, football, soccer:<br>
-</code>
+Note:This is a beta demo of GPT4All.
```
Comment (ParisNeo) [inline static/js/chat.js:246]: I need this to stay, I'll be moving it to a special section for examples

**Review by ParisNeo — COMMENTED** [review]
Please adress the issues reviewed in this pull request

**Comment by Jiankun-Huang** [comment]
Okay

**Comment by ParisNeo** [comment]
Hi there,

The project moved to nomic-ai's repo: https://github.com/nomic-ai/gpt4all-ui

Please use that repo from now on as this one will not be the official version.

I have made an issue about adding a ranking system there. If you want to contribute feel free to create a fork on that repo.

Best regards

**Pull request closed without merging**